src/components/FilterItem/FilterItem.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  ChangeEvent,
+  FC,
+  PropsWithChildren,
+  SyntheticEvent,
+  useState,
+} from "react";
+import cn from "../../utils/style";
+import { useFilterStore } from "../../context/FilterContext";
+import { useClickOutside } from "../../hooks/useClickOutside";
+
+export type FilterItemProps = PropsWithChildren & {
+  label: string;
+  name: string;
+  className?: string;
+};
+
+const FilterItem: FC<FilterItemProps> = ({ name, label, className }) => {
+  const [, addFilter] = useFilterStore();
+  const [isActive, setIsActive] = useState(false);
+  const [inputValue, setInputValue] = useState("");
+
+  const ref = useClickOutside(() => setIsActive(false));
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  const handleSubmit = (e: SyntheticEvent) => {
+    e.preventDefault();
+    addFilter(name, inputValue);
+  };
+
+  return (
+    <div ref={ref}>
+      <div
+        onClick={() => setIsActive(!isActive)}
```

Comment:
use callback fuction instead of oposit value of "current"  state

`setIsActive((oldActive) => !oldActive)`